modules/kafka/kafka-dest-worker.c

```diff
@@ -54,18 +54,13 @@ static gboolean
 _publish_message(KafkaDestWorker *self, LogMessage *msg)
 {
   KafkaDestDriver *owner = (KafkaDestDriver *) self->super.owner;
-  GString *key = self->key ? : &((GString)
-  {
-    .str = NULL,
-    .len = 0
-  });
   GString *message = self->message;
 
   if (rd_kafka_produce(owner->topic,
                        RD_KAFKA_PARTITION_UA,
                        RD_KAFKA_MSG_F_FREE | RD_KAFKA_MSG_F_BLOCK,
                        message->str, message->len,
-                       key->str, key->len,
+                       self->key->str, self->key->len,
```

Comment:
Key being NULL might have a significance here

Reply:
They key is now always alloced with `g_string_sized_new(0);`.

Reply:
But that would set str to a string with zero length, but key->str will never be NULL.

Reply:
Oh sorry, I misunderstood your comment.

Reply:
You are right. It comes to this in the end:
```
	if (key) {
		rkm->rkm_key     = p;
		rkm->rkm_key_len = keylen;
		memcpy(rkm->rkm_key, key, keylen);
	} else {
		rkm->rkm_key = NULL;
		rkm->rkm_key_len = 0;
	}
```
I will fix it!

Reply:
Changed it to `self->key->len ? self->key->str : NULL`